File: peer_lib/java/org/koalaui/arkoala/PeerNode.java
package org.koalaui.arkoala;

import java.util.Map;

public class PeerNode {
    public NativePeerNode peer;
    public PeerNode(ArkUINodeType type, int flags) {
        // TODO: rework ptr
        this.peer = new NativePeerNode(42, NativeModule._GetNodeFinalizer());
    }
    void applyAttributes(Map<String, Object> attrs) {}
}

File: peer_lib/koala-ui/arkoala-arkui/src/PeerNode.ts
import { int32 } from "@koalaui/common"
import { pointer } from "@koalaui/interop"
import { Finalizable } from "./Finalizable"
import { nativeModule } from "@koalaui/arkoala";
import { IncrementalNode } from "@koalaui/runtime"

export class NativePeerNode extends Finalizable {
}

const PeerNodeType = 11

export class PeerNode extends IncrementalNode {
    peer: NativePeerNode
    constructor(type: number, flags: int32) {
        super(PeerNodeType)
        const id = 0 // TODO: use id
        const ptr = nativeModule()._CreateNode(type, id, flags)
        this.peer = new NativePeerNode(ptr, getNodeFinalizer())
    }
    applyAttributes(attrs: Object) {}
}


function getNodeFinalizer() : pointer {
    return nativeModule()._GetNodeFinalizer()
}
